lib/trace-writer.js

```diff
@@ -83,12 +83,16 @@ TraceWriter.prototype.queueTrace_ = function(trace) {
     if (err) { return; } // ignore as index.js takes care of this.
 
     trace.projectId = project;
-    var length = that.buffer_.push(JSON.stringify(trace));
 
-    that.logger_.info('queued trace. new size:', length);
+    if (that.buffer_.length <= that.config_.bufferSize) {
+      that.buffer_.push(JSON.stringify(trace));
+      that.logger_.info('queued trace. new size:', that.buffer_.length);
```

Comment:
Change this to be logger_.debug. This is going to be too high volume to be enabled at info level.